.github/workflows/checks-definition.yml

```diff
@@ -0,0 +1,62 @@
+on:
+  pull_request
+
+jobs:
+  check-definitions:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout the repository
+      - name: Checkout repository
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+        with:
+          fetch-depth: 0  # Fetch all history for tags
+
+      # Fetch the latest tag
+      - name: Get latest tag
+        id: get_latest_tag
+        run: echo "LATEST_TAG=$(git describe --tags $(git rev-list --tags --max-count=1))" >> $GITHUB_ENV
+
+      # Compare definitions using git diff
+      - name: Compare API definitions
+        id: compare_definitions
+        run: |
+          # Get the diff between the latest tag and the current branch
+          git diff ${{ env.LATEST_TAG }} -- definitions.yml > definitions.diff
+
+          # Check if the diff file is not empty
+          if [ -s definitions.diff ]; then
+            echo "Changes detected in API definitions:"
+            cat definitions.diff
+
+            # Check for removed required values
+            if grep -qE '^-[[:space:]]+-' definitions.diff; then
```

Comment:
This regex also handles `x-extensible-enum` value removal, as seen [here](https://github.com/pagopa/io-services-metadata/actions/runs/14328799530/job/40159801142). What do you think about keeping this in scope?

Reply:
```diff
diff --git a/definitions.yml b/definitions.yml
index 12c21a49..9ca87f24 100644
--- a/definitions.yml
+++ b/definitions.yml
@@ -237,7 +237,6 @@ definitions:
     x-extensible-enum:
       - visible
       - hidden
-      - experimental
     example: visible
   BonusUrl:
     type: object
```

Reply:
I reverted to the previous workflow version to not over-complicate it. Of course we know that it's not bullet proof but it's a first shield towards human error. It cover 99% of our cases and for the 1% of the missing and false positive  ons we can handle them by hand. What do you think?